hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -112,20 +112,26 @@ public TaskResult reprocess(OMMetadataManager omMetadataManager) {
     for (String tableName : tables) {
       Table table = omMetadataManager.getTable(tableName);
 
-      try (TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator
-               = table.iterator()) {
+      try {
         if (tableHandlers.containsKey(tableName)) {
-          Triple<Long, Long, Long> details =
-              tableHandlers.get(tableName).getTableSizeAndCount(iterator);
-          objectCountMap.put(getTableCountKeyFromTable(tableName),
-              details.getLeft());
-          unReplicatedSizeMap.put(
-              getUnReplicatedSizeKeyFromTable(tableName), details.getMiddle());
-          replicatedSizeMap.put(getReplicatedSizeKeyFromTable(tableName),
-              details.getRight());
+          try (TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator
+                   = table.iterator()) {
+            Triple<Long, Long, Long> details =
+                tableHandlers.get(tableName).getTableSizeAndCount(iterator);
+            objectCountMap.put(getTableCountKeyFromTable(tableName),
+                details.getLeft());
+            unReplicatedSizeMap.put(
+                getUnReplicatedSizeKeyFromTable(tableName), details.getMiddle());
+            replicatedSizeMap.put(getReplicatedSizeKeyFromTable(tableName),
+                details.getRight());
+          }
         } else {
-          long count = Iterators.size(iterator);
-          objectCountMap.put(getTableCountKeyFromTable(tableName), count);
+          try (
+              TableIterator<String, ? extends Table.KeyValue<String, ?>> rawTableIterator
+                  = table.getRawTable().iterator()) {
+            long count = Iterators.size(rawTableIterator);
```

Comment:
Isn't `getEstimatedKeyCount()` good enough to get the count?

If not: instead of exposing the raw table from `TypedTable`, I think `getKeyCount()` should be implemented in `Table`, to which `TypedTable` could delegate.

Reply:
Thanks @adoroszlai for the review. `getEstimatedKeyCount`() doesn't provide accurate count. So we need iterator for table.

Reply:
> `getEstimatedKeyCount()` doesn't provide accurate count.

I know, the question was: does Recon _really_ need exact count?

And _if_ it does, then:

- Do not expose `Table<byte[], byte[]> getRawTable()` in the interface, as it breaks encapsulation.  It also doesn't match the existing interface, there is no mention of `byte[]` anywhere in `Table` (other than outdated comments).
- Add `getExactKeyCount()`, similar to `getEstimatedKeyCount()`, where `RDBTable` can has the logic, and other implementations can delegate to the underlying table.

Reply:
Ok understood your point and thank you for detailed explanation. I have done the changes. Kindly re-review.